Dockerfile

```diff
@@ -17,14 +18,14 @@ RUN bundle exec middleman build --build-dir=../public
 
 FROM ruby:3.4.2-alpine3.20
 
-RUN apk add --no-cache libxml2 yaml-dev
+RUN apk update && apk add --no-cache libxml2=2.12.7-r3 yaml-dev
```

Comment:
apk update is run before installs to ensure you get the latest patched packages.